Assets/BossRoom/Scripts/Client/Game/Character/CharacterSwap.cs

```diff
@@ -107,20 +159,47 @@ public void SwapToModel(int modelIndex)
                     m_Animator.runtimeAnimatorController = m_OriginalController;
                 }
             }
+
+            // lastly, now that we're all assembled, apply any override material.
+            switch (specialMaterialMode)
+            {
+                case SpecialMaterialMode.StealthySelf:
+                    SetOverrideMaterial(modelIndex, m_StealthySelfMaterial);
+                    break;
+                case SpecialMaterialMode.StealthyOther:
+                    SetOverrideMaterial(modelIndex, m_StealthyOtherMaterial);
+                    break;
+            }
         }
 
-        /// <summary>
-        /// Used by special effects where the character should be invisible.
-        /// </summary>
-        public void SwapAllOff()
+        private void ClearOverrideMaterial()
         {
-            for (int i = 0; i < m_CharacterModels.Length; i++)
+            foreach (var entry in m_OriginalMaterials)
             {
-                m_CharacterModels[i].SetFullActive(false);
-                if (m_CharacterModels[i].specialFx)
+                if (entry.Key)
                 {
-                    m_CharacterModels[i].specialFx.enabled = false;
+                    entry.Key.material = entry.Value;
+                }
+            }
+            m_OriginalMaterials.Clear();
+        }
+
+        private void SetOverrideMaterial(int modelIdx, Material overrideMaterial)
+        {
+            ClearOverrideMaterial(); // just sanity-checking; this should already have been called!
+            foreach (var bodyPartGO in m_CharacterModels[modelIdx].GetAllBodyParts())
+            {
+                if (!bodyPartGO)
+                {
+                    continue;
+                }
+                var renderer = bodyPartGO.GetComponent<Renderer>();

```

Comment:
Rename this variable here: Local variable 'renderer' hides property 'Component UnityEngine.Component.renderer'.

Reply:
ok